workflow/scripts/plot-volcano.R

```diff
@@ -0,0 +1,233 @@
+  # plot_data$log2FC[which(is.na(plot_data$log2FC))] = 0
+  # 
+  # plot_data$pval[which(is.nan(plot_data$pval))] = Inf
+  # plot_data$pval[which(is.na(plot_data$pval))] = 0
+  # 
+  # plot_data$qval[which(is.nan(plot_data$qval))] = Inf
+  # plot_data$qval[which(is.na(plot_data$qval))] = 0
+  
+  # cond1 <- df$sub == 1 & df$day == 2
+  # 
+  # cond2 <- df$sub == 3 & df$day == 4
+  # 
+  # df <- df[!(cond1 | cond2),]
+  # 
+  # is.na(plot_data$log2FC)
+  
+
+  #plot_data <- drop_na(plot_data)
+  
+  plot_data <- plot_data %>%
+    mutate(colour=factor(case_when(qval<.05 ~"most significant",
+                            abs(log2FC)>1 ~"significant",
+                            (qval<.05 && abs(log2FC)>1) ~"low significant",
+                            (qval>.05 || abs(log2FC)<1)~"not significant")))
+  interc_x <- c(-2,2) 
+  interc_y <- c(0.5,1)
+  max_x <- max(abs(plot_data$log2FC[is.finite(plot_data$log2FC)]), na.rm=T)
+  max_y <- max(abs(plot_data$pval[is.finite(plot_data$pval)]), na.rm = T)
+  if(is.numeric(snakemake@params[["range_log2FC"]])){
+    max_x <- snakemake@params[["range_log2FC"]]
+    interc_x <- c(-max_x/4, max_x/4)
+  }
+  
+  if(is.numeric(snakemake@params[["range_pval"]])){
+    max_y <- snakemake@params[["range_pval"]]
+    interc_y <- c(0, max_y/4)
+  }
+   # print(max_x)
+   # print(max_y)
+   
+   # print(plot_data$log2FC)
+   # print(plot_data$pval)
+
+  #---plot example for testing snakemakerule---
+  #pdf(file = snakemake@output[[1]], width = 14)
+  #example <- c(1, 2, 3, 7, 4, 9, 5)
+  #plot(example)
+  #dev.off()
+  #--------------------------------------------
+  write_tsv(fold_change, snakemake@output[["foldchange"]])
+  pdf(file = snakemake@output[["volcanoplot"]], width = 14)
+  #par()
+
+  #pdf(file = "results/plots/volcano/model_X.volcano.png", width = 14)
+    volcano <- ggplot(plot_data, aes(x=plot_data$log2FC, y=plot_data$pval))+
+      geom_point(aes(color=plot_data$colour)) +
+      ggtitle("Volcano-Plot")+
+      xlab("log2 fold change")+
+      ylab("p-values as -log10 adjusted")+
+      xlim(-max_x, max_x)+
+      ylim(0, max_y)+
+      scale_color_manual(name="significance",
+                         values = c("most significant"="red",
+                                    "significant"="orange",
+                                    "low significant"="green",
+                                    "not significant"="black"))+
+      geom_vline(xintercept = interc_x, color="blue", alpha=0.5)+
+      geom_hline(yintercept = interc_y, color="red", alpha=0.5)+
+      geom_vline(xintercept = 0, color="red", alpha=1)
+  print(volcano)
+  #---plot example for testing snakemakerule---
+  #pdf(file = snakemake@output[[1]], width = 14)
+  #example <- c(1, 2, 3, 7, 4, 9, 5)
+  #plot(example)
+  #dev.off()
+  #--------------------------------------------
+  # ggsave(filename=snakemake@output[[1]], width = 14)
+# #  write_tsv("foldchange", snakemake@output[["foldchange"]])
+   # print(volcano)
+  dev.off()
```

Comment:
I would opt for the more modern [`ggsave()`](https://ggplot2.tidyverse.org/reference/ggsave.html) function from the comments, here.